vignettes/geohashTools.Rmd

```diff
@@ -0,0 +1,155 @@
+
+```{r yirgacheffe}
+gh_decode('sc54v')
+```
+
+It can also be helpful to know just how precisely we've identified these coordinates; the `include_delta` argument gives the cell half-widths in both directions in addition to the cell centroid:
+
+```{r yirgacheffe_delta}
+gh_decode('sc54v', include_delta = TRUE)
+```
+For more detail on precision, see the table earlier on this vignette which shows the approximate level potential delta at different precision levels. 
+
+In terms of latitude and longitude, all geohashes with the same precision have the same dimensions (though the physical size of the "rectangle" changes depending on the latitude); as such it's easy to figure out thecell half-widths from the precision alone using `gh_delta`:
+
+```{r gh_delta}
+gh_delta(5L)
+```
+
+You can also pass entire vectors into `gh_decode` to decode multiple geohashes at once.  
+```{r decode_vector}
+gh_decode(gh)
+```
+
+## Geohash neighborhoods
+
+One unfortunate consequence of the geohash system is that, while geohashes that are lexicographically similar (e.g. `wxyz01` and `wxyz12`) are certainly close to one another, the converse is not true -- for example, `7gxyru` and `k58n2h` are neighbors! Put another way, small movements on the globe occasionally have visually huge jumps in the geohash-encoded output.
+
+The `gh_neighbors` function is designed to address this. Calling `gh_neighbours` will return all of the geohashes adjacent to a given geohash (or vector of geohashes) at the same level of precision. 
+
+For example, the Merlion statue in Singapore is roughly at `w21z74nz`, but this level of precision zooms in a bit too far. The geohash neighborhood thereof can be found with:
+
+```{r neighbors}
+gh_neighbors('w21z74nz')
+```
+
+## Working with spatial formats
+
+There are several `geohashTools` helper functions for converting geohashes into `sp` and `sf` class objects.
+
+### gh_to_*
+
+The `gh_to_sp`, `gh_to_spdf` and `gh_to_sf` functions convert geohash or vector of geohashes into spatial objects of `sp` and `sf` class respectively. Consider the previous example with the Singapore Merlion and some random data.
+```{r merlion_nbhd, fig.width = 6, fig.height = 6, out.width = '\\textwidth'}
+library(sf)
+merlion_ghs <- gh_neighbors("w21z74")
+merlion_nbhd <- gh_to_sf(merlion_ghs)
+plot(merlion_nbhd)
+```
+
+### gh_covering
+
+Sometimes we have a set of spatial datapoints that we want to aggregate or index using geohashes. The `gh_covering` function produces a grid of geohashes that overlap with the spatial points. For this function to work, the spatial object has to be in the [WGS84 (EPSG 4326)](https://epsg.io/4326) coordinate reference system that the geohashes use. 
+
+Let's use the included `meuse` dataset in the `sp` package which shows point coordinates for metal deposits. The first element in this object shows values for cadmium deposits. 
+```{r meuse, fig.width = 6, fig.height = 6, out.width = '\\textwidth'}
+library(ggplot2)
```

Comment:
I think we should treat this more as a suggested package, and branch out like

```
if (!requireNamespace(ggplot2)) {
  # either attempt to install it, or stop()
}

Reply:
> Travis failed:
> 
> > there is no package called 'ggplot2'
> 
> We need to upgrade the `r_packages` in `.travis.yml`

Fixed in a3c4fde

Reply:
> I think we should treat this more as a suggested package, and branch out like
> 
> ```
> if (!requireNamespace(ggplot2)) {
>   # either attempt to install it, or stop()
> }
> ```

Added in 26db6f5